### PR TITLE
New Markdown Plugin

### PR DESCRIPTION
**Pull request by gergely-ujvari**
This PR is a continuation of https://github.com/openannotation/annotator/pull/471
The markdown plugin is rewritten and simplified and can be inserted to a viewer as `renderText` option.

### CODE REVIEW

**Comment by tilgovi** [comment]
I think this look okay at a glance.

File: src/ui/markdown.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    _t = Util.gettext;
+
+function createMarkdownPlugin () {
+    var converter = null;
+
+    if (typeof g.Showdown !== "undefined" &&
+        g.Showdown !== null &&
+        typeof g.Showdown.converter !== "undefined" &&
+        g.Showdown.converter !== null) {
+        converter = new g.Showdown.converter();
```
Comment (aron) [inline src/ui/markdown.js]: Just out of curiosity, is there a case where Showdown can be defined without a `converter` method? Is this a versioning thing?

Comment (aron) [inline src/ui/markdown.js]: Looking at the history, I added that check, but didn't explain why. Sorry. Looking at the Showdown docs this function seems to be part of the API so perhaps a better check would be.

```
if (g.Showdown && typeof g.Showdown.converter === 'function') {
```

Comment (gergely-ujvari) [inline src/ui/markdown.js]: @aron: Changed the expression.

**Comment by gergely-ujvari** [comment]
Anyone care to merge?

**Comment by tilgovi** [comment]
I rebased to master, made it so it falls back to escaping if Showdown isn't loaded, and added it to the deafult UI plugin.

**Comment by tilgovi** [comment]
If you're happy, merge it!

**Pull request merged**